seaborn/categorical.py

```diff
@@ -1797,78 +1799,72 @@ def __init__(self, x, y, hue, data, order, hue_order,
                  width, dodge, k_depth, linewidth, scale, outlier_prop,
                  showfliers=True):
 
-        # TODO assigning variables for None is unneccesary
-        if width is None:
-            width = .8
         self.width = width
-
         self.dodge = dodge
-
-        if saturation is None:
-            saturation = .75
         self.saturation = saturation
 
-        if k_depth is None:
-            k_depth = 'proportion'
+        if k_depth not in ('proportion', 'tukey', 'trustworthy') and (
+            not isinstance(k_depth, numbers.Number)
+        ):
+            raise ValueError(f'k_depth "{k_depth}" not recognized')
```

Comment:
```suggestion
k_depth_methods = ["proportion", "tukey", "trustworthy"]
if k_depth not in k_depth_methods and not isinstance(k_depth, Number):
    msg = (f"k_depth must be one of {k_depth_methods} or a number, "
           f"but {k_depth} was passed.")
    raise ValueError(msg)
```

Reply:
(Requires direct import of `Number` I guess)

Reply:
Thanks, will change this.